server/src/main/java/org/elasticsearch/repositories/RepositoriesThrottlingStats.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.repositories;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.xcontent.ToXContentFragment;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import static java.util.stream.Collectors.toMap;
+
+public class RepositoriesThrottlingStats implements Writeable, ToXContentFragment {
```

Comment:
Maybe we can be a bit more general here and name this class `RepositoriesStats`?

Reply:
Actually there is already thing called `RepositoryStats` which is intended to hold request stats from underlying blob storage. Throttle metrics are modelled in Repository interface as a standalone entities, that's why I do not want to mix them with noted above blob store stats. Instead I'm creating stats class which wraps throttling metics only. Hence naming is quite specific.